### ISSUE TITLE
Update collapsed panel size when collapseSize changes

### ISSUE THREAD

**Issue by viveleroi** [bvaughn/react-resizable-panels#266]
We use a set pixel value for the collapsedSize of a panel. We convert the pixel size we want into a percentage, given the library needs percentages. One drawback from this is that when the window size changes, our `navCollapseSize` percentage does too - except only the initial collapse size is used.

It would be ideal if the library could detect changes to the `collapseSize` value and if the panel is collapsed, update the current size to match.

Or, add a setSize method to the `Panel` API so that I can set the size of this specific panel.

We have a navigation "drawer" component that is used in many different places in our and because we can't have it in a single shared root component, the `Panel` is defined here and not actually at the same level as the `PanelGroup`. So I believe the only way I can currently fix this is to move the size calculation logic to the panel group level but that's not going to work because the group isn't guaranteed to have this specific panel - it's up to the route. I'd have to split logic up.

```ts
const navMinSize = round(300 / width, 2) * 100
const navCollapseSize = round(36 / width, 2) * 100

<Panel
  className='flex'
  collapsedSize={navCollapseSize}
  collapsible
  defaultSize={navMinSize}
  id={`${side}Drawer`}
  maxSize={60}
  minSize={navMinSize}
  order={side === 'left' ? 1 : 3}
  ref={ref}
  onCollapse={() => {
    setIsCollapsed(true)
  }}
  onExpand={() => {
    setIsCollapsed(false)
  }}>
...
</Panel>
```

**Comment by bvaughn** [comment]
> Or, add a setSize method to the `Panel` API so that I can set the size of this specific panel.

This already exists (`resize`). You can check out the other imperative methods here:
https://react-resizable-panels.vercel.app/examples/imperative-panel-api

**Comment by bvaughn** [comment]
It may be worth clarifying that I intentionally decided **not** to support pixel sizes. If the library has bugs, I'll try to fix them, but if it's something that only happens due to trying to add pixel size constraints on top of the system, then I think that falls outside of the scope of what I can reasonably maintain as a side project (especially given that I don't need/use pixel constraints).

**Comment by viveleroi** [comment]
Oh I missed there was a resize method, I was looking for a setSize given there's a getSize. Also the onCollapse signature on the docs page is out of date.

**Comment by bvaughn** [comment]
> Oh I missed there was a resize method, I was looking for a setSize given there's a getSize.

The API is called `resize`.

It sounds like maybe this is a bug that should be fixed though, regardless of pixels vs percentage, so I'll leave it open.

> Also the onCollapse signature on the docs page is out of date.

I don't think this is mentioned on the docs.

That's also unrelated to the issue we're discussing though, so let's preferably keep this focused to one thing. (It's hard to track otherwise.)

**Comment by bvaughn** [comment]
I believe 0a410c8 should fix what I would consider an edge case bug with non-pixel usage (but one you'll need for pixel stuff to work). Hope it helps!

**Comment by bvaughn** [comment]
Released in react-resizable-panels@1.0.9

---

❤️ → ☕ [givebrian.coffee](http://givebrian.coffee/)

**Comment by viveleroi** [comment]
Thanks, this seems to work. I don't have to call `resize` anymore. Although one issue I noticed is that when you resize the window it randomly loses the collapsed state. I'm not sure why. Before you released 1.0.9 I had the following code to "force" it and I realized that I had to re-call the collapse method too, using my own isCollapsed state duplicate. Somehow that state was being lost, I thought maybe it was because as you resize the size becomes larger than the minSize which might trigger expand logic, but it happens even when I shrink the window size.

```ts
  useEffect(() => {
    const panel = ref.current
    if (panel && isCollapsed) {
      panel.resize(navCollapseSize)
      panel.collapse()
    }
  }, [navCollapseSize, isCollapsed])
```
